File: front/src/model/eventConfig.ts
export interface EventConfig {
    id: number;
    event_type: string;
    session_id?: number | null;
    enabled: boolean;
    frequency_minutes?: number | null;
}

export class EventConfigModel {
    public id: number;
    public event_type: string;
    public session_id?: number | null;
    public enabled: boolean;
    public frequency_minutes?: number | null;

    constructor() {
        this.id = 0;
        this.event_type = "";
        this.session_id = null;
        this.enabled = true;
        this.frequency_minutes = null;
    }

    static fromParsed(parsed: { id: number; event_type: string; session_id?: number | null; enabled: boolean; frequency_minutes?: number | null }): EventConfigModel {
        let config = new EventConfigModel();
        config.id = parsed.id;
        config.event_type = parsed.event_type;
        config.session_id = parsed.session_id ?? null;
        config.enabled = parsed.enabled;
        config.frequency_minutes = parsed.frequency_minutes ?? null;
        return config;
    }
}

File: back/src/entity/event_config.py
from sqlalchemy import Column, Integer, String, Boolean, ForeignKey
from sqlalchemy.orm import relationship

from back.src.driver.database import BaseModel
from back.src.entity.mixin import SerializableMixin


class EventConfig(SerializableMixin, BaseModel):
    __tablename__ = "event_config"

    id = Column(Integer, primary_key=True)
    event_type = Column(String, nullable=False)
    session_id = Column(Integer, ForeignKey("session.id"), nullable=True)  # NULL for global defaults
    enabled = Column(Boolean, nullable=False, default=True)
    frequency_minutes = Column(Integer, nullable=True)  # Optional frequency configuration

    session = relationship("RaclottoSession", lazy="selectin")






